src/licenses/LicenseModel.cpp

```diff
@@ -0,0 +1,184 @@
+#include "LicenseModel.h"
+
+#include <QDirIterator>
+
+static const char *const platformRootPath = ":/licenses";
+static const char *const qtRootPath = ":/licenses/qt";
+static const char *const toolkitRootPath = ":/licenses/qt/licenses";
+static const char *const webviewRootPath = ":/licenses/qt/licenses/qtwebengine/src/3rdparty/chromium";
+
+namespace licenses {
+
+  LicenseModel::LicenseModel(QObject *parent) : QAbstractItemModel(parent), m_dir(platformRootPath) {
+    populate();
+  }
+
+  LicenseModel::~LicenseModel() = default;
+
+  void LicenseModel::populate() {
+    QList<QVariant> root_categories;
+    QVector<QVariant> rootData = {QString("Licenses"), QString(platformRootPath)};
+    rootItem = std::make_unique<LicenseItem>(rootData, root_categories);
+
+    QHash<QString, LicenseItem *> map;
+    map[rootItem->path()] = rootItem.get();
+
+    QDirIterator it(m_dir, QDirIterator::Subdirectories);
+
+    while (it.hasNext()) {
+      QString path = it.next();
+
+      QList<QVariant> categories;
+      categories.append(QVariant(static_cast<int>(LicenseCategory::All)));
+
+      if (path.startsWith(webviewRootPath))
+        categories.append(QVariant(static_cast<int>(LicenseCategory::WebView)));
+
+      if (path.startsWith(toolkitRootPath) && !path.startsWith(webviewRootPath))
+        categories.append(QVariant(static_cast<int>(LicenseCategory::Toolkit)));
+
+      if (path.startsWith(platformRootPath) && !path.startsWith(qtRootPath))
```

Comment:
so there is a possible relation between toolkitRootPath and webviewRootPath (I assume, based on the `&& !path.startsWith(webviewRootPath)` above)
but there is no relationship between them and platformRootPath?  But there is a relationship between platform and qtRoot?

Maybe a comment somewhere here that explains "X might be a subfolder of Y and if so ... otherwise ..."

In general, I suspect future-you might appreciate at least one comment somewhere in all this code, even though the code is mostly clear without any.

Reply:
In a filesystem way platformRootPath contains (qtRootPath contains (toolkitRootPath contains webviewRootPath)

In a string way platformRootPath is a prefix of qtRootPath which is a prefix of toolkitRootPath which is a prefix of webviewRootPath

Reply:
So do you want to start with the most specific and work outwards?

    if (...webviewRootPath...)
         ...
    else if (...toolkitRootPath...)
        ...
    else if (...qtRootPath...)
    etc

?
(And with a comment explaining why)

Reply:
There is definitely something here I need to work out - I'll look at it this evening

Reply:
Ok, gave it a go, I think it's better now